pkg/endpoint/policy.go

```diff
@@ -267,12 +268,66 @@ func (e *Endpoint) regeneratePolicy(stats *regenerationStatistics, datapathRegen
 	// DistillPolicy converts a SelectorPolicy in to an EndpointPolicy
 	stats.endpointPolicyCalculation.Start()
 	result.endpointPolicy = selectorPolicy.DistillPolicy(e.getLogger(), e, desiredRedirects)
+	if pcr.NeedsRelease {
+		oldPolicy.Detach()
+	}
 	stats.endpointPolicyCalculation.End(true)
 
 	datapathRegenCtxt.policyResult = result
 	return nil
 }
 
+func (e *Endpoint) waitForPolicyComputationResult(
+	datapathRegenCtxt *datapathRegenerationContext,
+	securityIdentity *identity.Identity,
+	result *policyGenerateResult,
+) (*compute.Result, error) {
+	var (
+		computeResult compute.Result
+		retries       int
+	)
+
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+
+done:
+	for {
+		if retries >= 3 {
+			datapathRegenCtxt.policyResult = result
+			return nil, errors.New("failed to fetch computed policy")
```

Comment:
Should we add the triggering security identity in the error message to ease debugging?